src/DayPicker.tsx

```diff
@@ -256,6 +263,118 @@ export function DayPicker(props: DayPickerProps) {
 
   const dataAttributes = getDataAttributes(props);
 
+  const rootRef = useRef<HTMLDivElement>(null);
+  const rootElementSnapshot = useRef<HTMLElement>(null);
+  const previousMonths = useRef(months);
+
+  useLayoutEffect(() => {
```

Comment:
This is nice - could we place it in a new `useTransition` and unit test it?

Reply:
Indeed, I'm thinking of extracting this to a custom hook to isolate this logic, I'll see how easy it is to unit test it.
FYI, [useTransition](https://react.dev/reference/react/useTransition) is a native hook from React so I'll pick another name.